fairlearn/reductions/_exponentiated_gradient/_lagrangian.py

```diff
@@ -11,11 +11,14 @@
 
 from ._constants import _PRECISION, _INDENTATION, _LINE
 
-from fairlearn.reductions._moments import ClassificationMoment
+from fairlearn.reductions._moments import ClassificationMoment, LossMoment
 
 logger = logging.getLogger(__name__)
 
 
+_MESSAGE_BAD_OBJECTIVE = "Objective needs to be of the same type as constraints."
+
+
 class _Lagrangian:
```

Comment:
Not for this PR, but there was the little matter of making sure that this is pickleable.... my PR (#662) is waiting on whether it's OK to ditch the callable for an estimator :-)